linkedList.js

```diff
@@ -0,0 +1,372 @@
+class Node {
+  constructor(data) {
+    this._data = data;
+    this.nextNode = null;
+  }
+
+  getData() {
+    return this._data;
+  }
+
+
+}
+
+class LinkedList {
+  constructor() {
+    this.head = null;
+  }
+
+  length() {
+    var currentNode = this.head;
+    var length = 0;
+
+    if(!currentNode) {
+      return null;
+    }
+
+    while(currentNode) {
+      length++;
+      currentNode = currentNode.nextNode;
+    }
+    return length;
+  }
+
+  insert(value) {
+    var node = new Node(value);
+    var currentNode = this.head;
+
+    if(!currentNode) {
+      this.head = node;
+
+      return node;
+    }
+
+    currentNode = this.head;
+    node.nextNode = currentNode;
+    this.head = node;
+
+    this._length++;
+    return node;
+  }
+
+  search(value) {
+    var currentNode = this.head;
+    if(!currentNode) {
+      return false;
+    }
+
+    while(currentNode.nextNode) {
+      if(currentNode.getData() === value) {
+        return true;
+      }
+      currentNode = currentNode.nextNode;
+    }
+
+    return false;
+  }
+
+  findMax() {
+    var currentNode = this.head;
+    var max = this.head.getData();
+    if(!currentNode) {
+      return null;
+    }
+
+    for(var i = 0; i < this.length(); i++) {
+      if(currentNode.getData() > max) {
+        max = currentNode.getData();
+      }
+      currentNode = currentNode.nextNode;
+    }
+
+    return max;
+  }
+
+  findMin() {
+    var currentNode = this.head;
+    var min = this.head.getData();
```

Comment:
Same comment as with findMax. Check for null before you access member variables in the node.